dbms/include/DB/AggregateFunctions/AggregateFunctionGroupUniqArray.h

```diff
@@ -94,11 +95,142 @@ class AggregateFunctionGroupUniqArray
 		data_to.resize(old_size + size);
 
 		size_t i = 0;
-		for (typename State::Set::const_iterator it = set.begin(); it != set.end(); ++it, ++i)
+		for (auto it = set.begin(); it != set.end(); ++it, ++i)
 			data_to[old_size + i] = *it;
 	}
 };
 
+
+/// Generic implementation
+struct AggreagteFunctionGroupUniqArrayGenericData
+{
+	using Set = HashSetWithSavedHash<StringRef, StringRefHash, HashTableGrower<4>, HashTableAllocatorWithStackMemory<16>>;
```

Comment:
Maybe there is inconsistency, because in HashTableGrower, parameter is power of two for space of hash table in number of elements, but in HashTableAllocatorWithStackMemory, parameter is number of bytes.
Please compare with other uses.